velox/connectors/hive/HiveConnector.h

```diff
@@ -39,17 +39,18 @@ class HiveConnector : public Connector {
       const std::unordered_map<
           std::string,
           std::shared_ptr<connector::ColumnHandle>>& columnHandles,
-      ConnectorQueryCtx* connectorQueryCtx) override {
+      ConnectorQueryCtx* connectorQueryCtx,
+      const dwio::common::ReaderOptions& options) override {
     return std::make_unique<HiveDataSource>(
         outputType,
         tableHandle,
         columnHandles,
         &fileHandleFactory_,
-        connectorQueryCtx->memoryPool(),
         connectorQueryCtx->expressionEvaluator(),
         connectorQueryCtx->allocator(),
         connectorQueryCtx->scanId(),
-        executor_);
+        executor_,
+        options);
```

Comment:
Don't we have to set `options->setMemoryPool(connectorQueryCtx->memoryPool())` before returning?

Reply:
I am not sure but I would expect the creator of the ReaderOptions to have done that already. In fact TableScan.cpp line 111 does just that.